amplitude/amplitude_client.go

```diff
@@ -0,0 +1,91 @@
+package amplitude
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+)
+
+func newAmplitudeClient(
+	serverURL string, options clientPayloadOptions, logger Logger, connectionTimeout time.Duration,
+) *amplitudeClient {
+	var payloadOptions *clientPayloadOptions
+	if options != (clientPayloadOptions{}) {
```

Comment:
Minor, I might be wrong, maybe just use `options` directly in `amplitudeClient`? I am not sure what is the difference.

Reply:
To not serialize empty options `options: {}` and save some payload size.